lib/class-wp-annotation-caps.php

```diff
@@ -0,0 +1,441 @@
+<?php
+/**
+ * Annotations API: WP_Annotation_Caps class
+ *
+ * @package gutenberg
+ * @since [version]
+ */
+
+/**
+ * Annotation capabilities.
+ *
+ * @since [version]
+ */
+final class WP_Annotation_Caps {
+	/**
+	 * Maps annotation caps.
+	 *
+	 * @since [version]
+	 *
+	 * @param array  $caps    Required capabilities.
+	 * @param string $cap     Capability to check/map.
+	 * @param int    $user_id User ID (empty if not logged in).
+	 * @param array  $args    Arguments to {@see map_meta_cap()}.
+	 *
+	 * @return array          Array of required capabilities.
+	 */
+	public static function on_map_meta_cap( $caps, $cap, $user_id, $args ) {
+		switch ( $cap ) {
+			case 'create_annotation':
+				return self::map_create_annotation( $caps, $cap, $user_id, $args );
+
+			case 'read_annotation':
+				return self::map_read_annotation( $caps, $cap, $user_id, $args );
+
+			case 'read_annotations':
+				return self::map_read_annotations( $caps, $cap, $user_id, $args );
+
+			case 'edit_annotation':
+				return self::map_edit_annotation( $caps, $cap, $user_id, $args );
+
+			case 'delete_annotation':
+				return self::map_delete_annotation( $caps, $cap, $user_id, $args );
+
+			case 'create_annotations':
+			case 'delete_annotations':
+			case 'delete_others_annotations':
+			case 'delete_private_annotations':
+			case 'delete_published_annotations':
+			case 'edit_annotations':
+			case 'edit_others_annotations':
+			case 'edit_private_annotations':
+			case 'edit_published_annotations':
+			case 'publish_annotations':
+			case 'read_private_annotations':
+				return self::map_others( $caps, $cap, $user_id, $args );
+		}
+
+		return $caps;
+	}
+
+	/**
+	 * Maps the 'create_annotation' meta-cap.
+	 *
+	 * Requires $args[0] with a post ID, and optionally supports $args[1] containing an
+	 * annotation comment type. If not given, comment type defaults to 'annotation'.
+	 * e.g., `current_user_can( 'create_annotation', 123 )`
+	 *
+	 * @since [version]
+	 *
+	 * @param array  $caps    Required capabilities.
+	 * @param string $cap     Capability to check/map.
+	 * @param int    $user_id User ID (empty if not logged in).
+	 * @param array  $args    Arguments to {@see map_meta_cap()}.
```

Comment:
Since this function forks itself out to the other cap-specific handlers, I documented the arguments separately in each one to help clarify. Thanks again!